--- demangle_orig.cc	2021-10-06 09:12:56.915195396 +0200
+++ demangle.cc	2021-10-05 09:07:08.625867843 +0200
@@ -167,7 +167,7 @@
 // Returns true if "str" has at least "n" characters remaining.
 static bool AtLeastNumCharsRemaining(const char *str, int n) {
   for (int i = 0; i < n; ++i) {
-    if (str == '\0') {
+    if (str == "\0") {
       return false;
     }
   }
@@ -223,7 +223,7 @@
 // Returns true and advances "mangled_cur" if we find any character in
 // "char_class" at "mangled_cur" position.
 static bool ParseCharClass(State *state, const char *char_class) {
-  if (state->mangled_cur == '\0') {
+  if (state->mangled_cur == "\0") {
     return false;
   }
   const char *p = char_class;
